tests/Unit/Evolution/Systems/ScalarWave/Test_Constraints.cpp

```diff
@@ -0,0 +1,351 @@
+  get<ScalarWave::Pi>(sw_vars) = pi;
+  get<ScalarWave::Phi<dim>>(sw_vars) = phi;
+  // Second, compute derivatives
+  const auto sw_derivs =
+      partial_derivatives<VariablesTags, VariablesTags, dim, Frame::Inertial>(
+          sw_vars, mesh, coord_map.inv_jacobian(x_logical));
+  const auto& deriv_psi =
+      get<Tags::deriv<ScalarWave::Psi, tmpl::size_t<dim>, Frame::Inertial>>(
+          sw_derivs);
+  const auto& deriv_pi =
+      get<Tags::deriv<ScalarWave::Pi, tmpl::size_t<dim>, Frame::Inertial>>(
+          sw_derivs);
+  const auto& deriv_phi = get<
+      Tags::deriv<ScalarWave::Phi<dim>, tmpl::size_t<dim>, Frame::Inertial>>(
+      sw_derivs);
+
+  // Insert into databox
+  const auto box = db::create<
+      db::AddSimpleTags<
+          ::Tags::Coordinates<dim, Frame::Inertial>, ScalarWave::Psi,
+          ScalarWave::Phi<dim>, ScalarWave::Pi,
+          ::Tags::deriv<ScalarWave::Psi, tmpl::size_t<dim>, Frame::Inertial>,
+          ::Tags::deriv<ScalarWave::Phi<dim>, tmpl::size_t<dim>,
+                        Frame::Inertial>,
+          ::Tags::deriv<ScalarWave::Pi, tmpl::size_t<dim>, Frame::Inertial>>,
+      db::AddComputeTags<ScalarWave::Tags::OneIndexConstraintCompute<dim>,
+                         ScalarWave::Tags::TwoIndexConstraintCompute<dim>>>(
+      x, psi, phi, pi, deriv_psi, deriv_phi, deriv_pi);
+
+  // Compute tested quantities locally
+  const auto one_index_constraint =
+      ScalarWave::one_index_constraint(deriv_psi, phi);
+  const auto two_index_constraint = ScalarWave::two_index_constraint(deriv_phi);
+
+  // Check that their compute items in databox furnish identical values
+  CHECK(db::get<ScalarWave::Tags::OneIndexConstraint<dim>>(box) ==
+        one_index_constraint);
+  CHECK(db::get<ScalarWave::Tags::TwoIndexConstraint<dim>>(box) ==
+        two_index_constraint);
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.ScalarWave.Constraints",
+                  "[Unit][Evolution]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "Evolution/Systems/ScalarWave/"};
+  // Test the one-index constraint with random numbers
+  test_one_index_constraint_random<1>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_one_index_constraint_random<2>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_one_index_constraint_random<3>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+
+  // Test the two-index constraint with random numbers
+  test_two_index_constraint_random<1>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_two_index_constraint_random<2>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_two_index_constraint_random<3>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+
+  // Test constraints against Spherical wave
+  const size_t grid_size = 8;
+  const std::array<double, 3> upper_bound{{0.82, 1.24, 1.32}};
```

Comment:
This domain appears to very small. Do the tests fail for something that would be more encompassing of the wave profile

Reply:
expanded